client/my-sites/stats/stats-purchase/index.tsx

```diff
@@ -96,12 +104,19 @@ const StatsPurchasePage = ( { query }: { query: { redirect_uri: string; from: st
 		( ! siteProducts && isRequestingSiteProducts );
 
 	const [ initialStep, initialSiteType ] = useMemo( () => {
+		// if the site is detected as commercial
+		if ( isTypeDetectionEnabled ) {
+			if ( options.isCommercial && ! isCommercialOwned ) {
+				return [ SCREEN_PURCHASE, TYPE_COMMERCIAL ];
+			}
+		}
+
 		if ( isPWYWOwned && ! isCommercialOwned ) {
 			return [ SCREEN_PURCHASE, TYPE_COMMERCIAL ];
 		}
 		// if nothing is owned don't specify the type
 		return [ SCREEN_TYPE_SELECTION, null ];
-	}, [ isPWYWOwned, isCommercialOwned ] );
+	}, [ isPWYWOwned, isCommercialOwned, options?.isCommercial, isTypeDetectionEnabled ] );
```

Comment:
```suggestion
	}, [ isPWYWOwned, isCommercialOwned, options.isCommercial, isTypeDetectionEnabled ] );
```

Could we remove the optional syntax for the `options` parameter since an empty object should be passed at least?